jerry-core/api/jerry.c

```diff
@@ -1482,6 +1483,86 @@ jerry_create_string_sz (const jerry_char_t *str_p, /**< pointer to string */
   return ecma_make_string_value (ecma_str_p);
 } /* jerry_create_string_sz */
 
+/**
+ * Calculates the size of the given pattern and creates a RegExp object.
+ *
+ * @return value of the constructed RegExp object.
+ */
+jerry_value_t
+jerry_create_regexp (const jerry_char_t *pattern,
+                     jerry_regexp_flags_t flags)
+{
+  return jerry_create_regexp_sz (pattern, lit_zt_utf8_string_size (pattern), flags);
+} /* jerry_create_regexp */
+
+/**
+ * Creates a RegExp object with the given pattern and flags.
+ *
+ * @return value of the constructed RegExp object.
+ */
+jerry_value_t
+jerry_create_regexp_sz (const jerry_char_t *pattern, /**< RegExp pattern */
+                        jerry_size_t pattern_size, /**< Length of the pattern */
+                        jerry_regexp_flags_t flags) /**< Optional RegExp flags */
+{
+  jerry_assert_api_available ();
+
+#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
+  if (!lit_is_valid_utf8_string (pattern, pattern_size))
+  {
+    return jerry_throw (ecma_raise_common_error ("Input must be a valid utf8 string"));
+  }
+
+  ecma_string_t *ecma_pattern = ecma_new_ecma_string_from_utf8 (pattern, pattern_size);
+
+  ecma_value_t str_val = ecma_make_string_value (ecma_pattern);
+
+  jerry_value_t ret_val;
+
+  if (flags > 0)
+  {
+    ecma_value_t flag_val = ECMA_VALUE_UNDEFINED;
+    jerry_char_t flag_str[3];
+    unsigned char *str_ptr = flag_str;
+
+    if (flags & JERRY_REGEXP_FLAG_GLOBAL)
+    {
+      *str_ptr++ = 'g';
+    }
+
+    if (flags & JERRY_REGEXP_FLAG_MULTILINE)
+    {
+      *str_ptr++ = 'm';
+    }
+
+    if (flags & JERRY_REGEXP_FLAG_IGNORE_CASE)
+    {
+      *str_ptr++ = 'i';
+    }
+
+    flag_val = jerry_create_string_sz (flag_str, (jerry_size_t) (str_ptr - flag_str));
+
+    ecma_value_t args[2] = {str_val, flag_val};
+    ret_val = ecma_builtin_regexp_dispatch_construct (args, 2);
```

Comment:
The latest update is still `ecma_builtin_regexp_dispatch_construct`-based. Change has been requested. See my previous comments for details.

Reply:
I know, it is still in progress, I just wanted to update the patch with the style issue fixes. Sorry for not informing you about that.